### PR TITLE
migrate GooglePubSubConsumingTest to new test framework

### PR DESCRIPTION
**Pull request by piotrrzysko**

### CODE REVIEW

File: integration-tests/src/slowIntegrationTest/java/pl/allegro/tech/hermes/integrationtests/AttachingKeepAliveHeaderTest.java [hunk 1]
```diff
@@ -21,23 +19,15 @@
 
 public class AttachingKeepAliveHeaderTest {
 
+    @Order(0)
```
Comment (szczygiel-m) [inline integration-tests/src/slowIntegrationTest/java/pl/allegro/tech/hermes/integrationtests/AttachingKeepAliveHeaderTest.java:22]: Are we sure we need to use this annotation? I thought the tool was smart enough to take care of the order itself

**Review by szczygiel-m — APPROVED** [review]

File: integration-tests/src/slowIntegrationTest/java/pl/allegro/tech/hermes/integrationtests/AttachingKeepAliveHeaderTest.java (hunk 1 above)
Comment (piotrrzysko) [reply integration-tests/src/slowIntegrationTest/java/pl/allegro/tech/hermes/integrationtests/AttachingKeepAliveHeaderTest.java:22]: I'm not sure, but to me, the contract of `@RegisterExtension` seems too vague to assume that `@Order` is unnecessary in this case: https://junit.org/junit5/docs/5.4.0/api/org/junit/jupiter/api/extension/RegisterExtension.html

File: integration-tests/src/slowIntegrationTest/java/pl/allegro/tech/hermes/integrationtests/GooglePubSubConsumingTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    @Order(2)
+    @RegisterExtension
+    public static HermesFrontendExtension frontend = new HermesFrontendExtension(infra);
+
+    public static final GooglePubSubContainer googlePubSubContainer = new GooglePubSubContainer();
+
+    public static HermesConsumersTestApp consumer;
+
+    @BeforeAll
+    public static void startPubSub() {
```
Comment (moscicky) [inline integration-tests/src/slowIntegrationTest/java/pl/allegro/tech/hermes/integrationtests/GooglePubSubConsumingTest.java:44]: nit: both beforeAll and afterAll are responsible for more than the method name would suggest (also consumers)

**Review by moscicky — APPROVED** [review]

File: integration-tests/src/slowIntegrationTest/java/pl/allegro/tech/hermes/integrationtests/GooglePubSubConsumingTest.java (hunk 2 above)
Comment (piotrrzysko) [reply integration-tests/src/slowIntegrationTest/java/pl/allegro/tech/hermes/integrationtests/GooglePubSubConsumingTest.java]: fixed

**Pull request merged**